### PR TITLE
Extracted test cases from SolidityEndToEnd.cpp

### PR DESCRIPTION
**Pull request by nishant-sachdeva**
Extracted Test Cases from SolidityEndToEnd.cpp are := 
1. enum_referencing ( removed in latest commit )
2. invalid_enum_logged
3. library_staticcall_delegatecall
4. memory_types_initialisation
5. constant_string_literal
6. return_string
7. packed_storage_structs_delete
8. storing_invalid_boolean

Helps #12253 . Extended from previous Draft PR at #12289

### CODE REVIEW

File: test/libsolidity/semanticTests/enums/enum_referencing.sol [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    function y() public pure returns (I.Direction) {
+        return L.g();
+    }
+}
+// ====
+// compileViaYul: also
+// compileToEwasm: false
+// ----
+// library: L
+// f() -> 3
```
Comment (chriseth) [inline test/libsolidity/semanticTests/enums/enum_referencing.sol]: Wasn't this 1 in the original test?

Comment (chriseth) [reply test/libsolidity/semanticTests/enums/enum_referencing.sol]: Ah ok now I see: The original test first calls a function on L itself. I'm not sure if we can do this inside the semantic tests.

File: test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            return 3;
+        delete x;
+        delete data;
+        return 1;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// test() -> 1
```
Comment (chriseth) [inline test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol:26]: Please add the storage empty check.

File: test/libsolidity/semanticTests/strings/return_string.sol [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        return s;
+    }
+    function get2() public returns (string memory r) {
+        r = s;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// get1() -> 0x20, 0
```
Comment (chriseth) [inline test/libsolidity/semanticTests/strings/return_string.sol]: This is missing `set(string): 0x20, 5, "Julia"`

**Comment by chriseth** [comment]
The ones I did not comment on look good.

File: test/libsolidity/semanticTests/strings/return_string.sol (hunk 3 above)
Comment (nishant-sachdeva) [reply test/libsolidity/semanticTests/strings/return_string.sol]: Update : Added `set(string)` function to the .sol file and pushed changes

File: Changelog.md [hunk 4]
```diff
@@ -4,6 +4,7 @@ Language Features:
 
 
 Compiler Features:
+* Extracted test cases from SolidityEndToEnd.cpp
```
Comment (chriseth) [inline Changelog.md]: We do not add a changelog entry if the change is not visible to the users.

**Review by chriseth — DISMISSED** [review]

File: Changelog.md (hunk 4 above)
Comment (nishant-sachdeva) [reply Changelog.md]: Removed in latest push.

File: test/libsolidity/semanticTests/enums/enum_referencing.sol (hunk 1 above)
Comment (nishant-sachdeva) [reply test/libsolidity/semanticTests/enums/enum_referencing.sol]: Removed from this commit and moved to another branch for further work.

File: test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol (hunk 2 above)
Comment (nishant-sachdeva) [reply test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol:26]: Added the checks needed.

File: test/libsolidity/semanticTests/enums/enum_referencing.sol (hunk 1 above)
Comment (nishant-sachdeva) [reply test/libsolidity/semanticTests/enums/enum_referencing.sol]: The commit has been extracted from SolidityEndToEnd.cpp 

Should we insert it back there as cpp code? Or should we merely shift the .sol file to a different folder

File: test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    function checkIfDataIsEmpty() public returns (bool) {
+        return (x == 0 && y == 0 && data.a == 0 && data.b == 0 && data.c == 0);
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// test() -> 1
+// checkIfDataIsEmpty() -> true
```
Comment (chriseth) [inline test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol]: Sorry, I should have told you there is a special mechanism for this:
```suggestion
// storageEmpty -> 1
```
Can you undo the changes to the contract itself?

Comment (nishant-sachdeva) [reply test/libsolidity/semanticTests/structs/packed_storage_structs_delete.sol]: Made these changes in the latest push.

**Review by chriseth — APPROVED** [review]
